tests/jobs.py

```diff
@@ -901,80 +901,6 @@ def do_schedule(function, function2):
         do_schedule()
         self.assertEquals(2, len(executor.queued_tasks))
 
-    def test_retry_still_in_executor(self):
```

Comment:
Ok, if that is the case (I can't look through the code now), the test needs to be fixed. Ie. the logic if the test is also sound: try a task retries times until it is finished / failed. 

To me it seems we might have a larger issue at hand: first retries were not executed (with a fix that is unrelated to dag runs) and now, maybe, we prematurely set dag_runs to failed. 

Maybe we should discuss this over a video call as I am not sure we are totally aligned in our understanding.